src/compressed_tensors/quantization/quant_config.py

```diff
@@ -250,3 +250,28 @@ def requires_calibration_data(self):
                     return True
 
         return False
+
+    def remove_defaults(self):
+        defaults = self.__fields__
+        result = {}
+        for field, value in dict(self).items():
+            if value is None:
+                continue
+            if field == "config_groups":
+                outer_value = value
+                value = {}
+                for config_key, config_value in outer_value.items():
+                    config_value = config_value.remove_defaults()
+                    value[config_key] = config_value
+
+            if field == "ignore" and len(value) == 0:
```

Comment:
Why can't we compare directly to the default value? This is inflexible if we want to change the ignore to default to a non-empty list